Products/CMFPlone/browser/iconresolver.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+from OFS.Image import File
+from plone.registry.interfaces import IRegistry
+from Products.CMFCore.interfaces import ISiteRoot
+from zope.component import adapter
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.location.interfaces import LocationError
+from zope.traversing.interfaces import ITraversable
+from zope.publisher.interfaces import IPublishTraverse
+
+
+@implementer(IPublishTraverse)
+class IconBaseResolver:
+
+    prefix = 'plone.staticresources.icon.'
+
+    def __init__(self, context, name):
+        self.context = context
+        self.name = name
+
+    def __call__(self):
+        registry = getUtility(IRegistry)
+        icon = prefix + self.name
+        if icon in registry:
+            return registry[icon]
+        raise LocationError(self.name)
+
+
+class IconURLResolver(IconBaseResolver):
+
+    def __call__(self):
+        value = super().__call__()
+        url = getSite().absolute_url() + '/' + value
```

Comment:
I think here you want the navigation root URL from plone portal state: https://github.com/plone/plone.app.layout/blob/5f597e41c413c1584a25f01347ead48da52d77c4/plone/app/layout/globals/portal.py#L67
Example: https://github.com/plone/Products.CMFPlone/blob/c3c7ec9ea52fde4d1e868f7df703a80cc8325c0a/Products/CMFPlone/browser/ploneview.py#L156-L160

Reply:
I'm not sure. URL is used to construct the src of e.g. an image-tag. I expcected absolute_url of site is right, when using ++resolveiconurl++some-icon.